test/e2e/operator_test.go

```diff
@@ -1284,3 +1469,97 @@ func stsModeRequested() bool {
 	}
 	return false
 }
+
+// extractArg extracts the value of a specific argument from a list of container arguments.
+// Returns an error if the argument is present but malformed, or if not found.
+func extractArg(args []string, argKey string) (string, error) {
+	for _, arg := range args {
+		if strings.HasPrefix(arg, argKey+"=") {
+			parts := strings.SplitN(arg, "=", 2)
+			if len(parts) != 2 {
+				return "", fmt.Errorf("invalid format for argument %s: %s", argKey, arg)
+			}
+			return parts[1], nil
+		}
+	}
+	return "", fmt.Errorf("argument %s not found", argKey)
+}
+
+// assertContainerArgFromDeployment asserts that a container in a deployment has a
+// specific argument with an expected value.
+func assertContainerArgFromDeployment(t *testing.T, dep *appsv1.Deployment, containerName, argKey, expectedArgValue string) {
+	t.Helper()
+	t.Logf("Asserting container %q has argument %q with value %q", containerName, argKey, expectedArgValue)
+	for _, c := range dep.Spec.Template.Spec.Containers {
+		if c.Name == containerName {
+			actualArgValue, err := extractArg(c.Args, argKey)
+			if err != nil {
+				t.Fatalf("failed to extract argument %q for container %q: %v", argKey, containerName, err)
+			}
+
+			if actualArgValue != expectedArgValue {
+				t.Fatalf("container %q, argument %q: expected value %q, but got %q", containerName, argKey, expectedArgValue, actualArgValue)
+			}
+			return
+		}
+	}
+	t.Fatalf("container %q not found in deployment", containerName)
+}
+
+// assertELBbyTags asserts that an ELB instance has the expected tags.
+func assertELBbyTags(t *testing.T, expectedTags map[string]string) {
+	t.Helper()
+	t.Logf("Asserting ELBs by tags %v", expectedTags)
+
+	rgtClient := resourcegroupstaggingapi.NewFromConfig(cfg)
+
+	err := wait.PollUntilContextTimeout(context.Background(), 30*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+		lbARNs, err := getLoadBalancerARNsByTags(ctx, rgtClient, expectedTags)
+		if err != nil {
+			return false, fmt.Errorf("unable to get ELBs for %v tags: %v", expectedTags, err)
+		}
+		// There must be exactly 1 ELB with the given tags
+		if len(lbARNs) != 1 {
+			t.Logf("expected single ELB with %v tags, but got %d (with arns: %v), retrying... ", expectedTags, len(lbARNs), lbARNs)
+			return false, nil
+		}
+
+		t.Logf("found ELB with %v tags (with arn: %v)", expectedTags, lbARNs[0])
+		return true, nil
+	})
+
+	if err != nil {
+		t.Fatalf("Timed out waiting for %v tags to match an ELB: %v", expectedTags, err)
```

Comment:
To match the other fatal messages:
```suggestion
		t.Fatalf("timed out waiting for %v tags to match an ELB: %v", expectedTags, err)
```

Reply:
Updated